Compilation fails (conflicting types) if mpfr-devel 4.2.0 is provided

Hello,

I tried compiling sleef 3.5.1 with mpfr 4.2.0 support on openSUSE but it failed with a bunch of conflicting type errors:

```
[   97s] /home/abuild/rpmbuild/BUILD/sleef-3.5.1/src/libm-tester/testerutil.h:93:6: error: conflicting types for 'mpfr_sinpi'; have 'void(__mpfr_struct *, __mpfr_struct *, mpfr_rnd_t)'
[   97s]    93 | void mpfr_sinpi(mpfr_t ret, mpfr_t arg, mpfr_rnd_t rnd);
[   97s]       |      ^~~~~~~~~~
[   97s] In file included from /home/abuild/rpmbuild/BUILD/sleef-3.5.1/src/libm-tester/tester2simddp.c:9:
[   97s] /usr/include/mpfr.h:752:21: note: previous declaration of 'mpfr_sinpi' with type 'int(__mpfr_struct *, const __mpfr_struct *, mpfr_rnd_t)'
[   97s]   752 | __MPFR_DECLSPEC int mpfr_sinpi (mpfr_ptr, mpfr_srcptr, mpfr_rnd_t);
[   97s]       |                     ^~~~~~~~~~
[   97s] /home/abuild/rpmbuild/BUILD/sleef-3.5.1/src/libm-tester/testerutil.h:94:6: error: conflicting types for 'mpfr_cospi'; have 'void(__mpfr_struct *, __mpfr_struct *, mpfr_rnd_t)'
[   97s]    94 | void mpfr_cospi(mpfr_t ret, mpfr_t arg, mpfr_rnd_t rnd);
[   97s]       |      ^~~~~~~~~~
[   97s] /usr/include/mpfr.h:753:21: note: previous declaration of 'mpfr_cospi' with type 'int(__mpfr_struct *, const __mpfr_struct *, mpfr_rnd_t)'
[   97s]   753 | __MPFR_DECLSPEC int mpfr_cospi (mpfr_ptr, mpfr_srcptr, mpfr_rnd_t);
```

If mpfr is not provided, however, sleef will be compiled successfully. Can I safely package sleef without mpfr support? Thank you.

--- a/src/libm-tester/testerutil.c
+++ b/src/libm-tester/testerutil.c
@@ -289,6 +289,7 @@ double countULP2sp(float d, mpfr_t c0) {
 
 //
 
+#if MPFR_VERSION < MPFR_VERSION_NUM(4, 2, 0)
 void mpfr_sinpi(mpfr_t ret, mpfr_t arg, mpfr_rnd_t rnd) {
   mpfr_t frpi, frd;
   mpfr_inits(frpi, frd, NULL);
@@ -314,6 +315,7 @@ void mpfr_cospi(mpfr_t ret, mpfr_t arg, mpfr_rnd_t rnd) {
 
   mpfr_clears(frpi, frd, NULL);
 }
+#endif
 
 void mpfr_lgamma_nosign(mpfr_t ret, mpfr_t arg, mpfr_rnd_t rnd) {
   int s;

--- a/src/libm-tester/testerutil.h
+++ b/src/libm-tester/testerutil.h
@@ -90,7 +90,9 @@ int cmpDenormsp(float x, mpfr_t fry);
 double countULPsp(float d, mpfr_t c);
 double countULP2sp(float d, mpfr_t c);
 
+#if MPFR_VERSION < MPFR_VERSION_NUM(4, 2, 0)
 void mpfr_sinpi(mpfr_t ret, mpfr_t arg, mpfr_rnd_t rnd);
 void mpfr_cospi(mpfr_t ret, mpfr_t arg, mpfr_rnd_t rnd);
+#endif
 void mpfr_lgamma_nosign(mpfr_t ret, mpfr_t arg, mpfr_rnd_t rnd);
 #endif

